core/src/main/scala/org/apache/spark/scheduler/EventLoggingListener.scala

```diff
@@ -249,21 +249,18 @@ private[spark] class EventLoggingListener(
   }
 
   override def onExecutorMetricsUpdate(event: SparkListenerExecutorMetricsUpdate): Unit = {
+    if (event.execId == SparkContext.DRIVER_IDENTIFIER) {
+      logEvent(event)
+    }
```

Comment:
See #31992 for discussion on this.
Log only when `shouldLogStageExecutorMetrics` is enabled.

Reply:
> 

previously, it's controlled by `shouldLogStageExecutorMetrics` because of `a single event for both driver and executor metrics update `.
firstly: `logEvent` of  driver has no  Inevitable relation with `a single event for both driver and executor metrics update `, that's two different things, we can log the driver event with `shouldLogStageExecutorMetrics`, and also control the update semantics by the parameter on the same time.
secondary: `shouldLogStageExecutorMetrics` is kind of heavy operation, i don't think it's a good idea to bind it with logging driver metric
thirdly:  i don't see the  worth  of driver  metric in `stageExecutorMetrics`, the driver metric in it is never be used.

Reply:
maybe we can just remove the driver metric updates in `stageExecutorMetrics`, or update it with the `shouldLogStageExecutorMetrics` while logging the driver event without on the same time.
anyway thank you so much for reviewing

Reply:
besides, in `AppStatusListener` the `updateStageLevelPeakExecutorMetrics`  never updates different stage executor metrics when accept a driver metric . maybe we should consistently apply it both in history side and appAlive side.

Reply:
I am not sure I follow the comment thread.

The current behavior is, `SparkListenerExecutorMetricsUpdate` is logged when `spark.eventLog.logStageExecutorMetrics` is enabled. I am not seeing good reasons to change this behavior.
Given event files are consumed not just by spark, but external applications and analysis tools, changing semantics  needs to be looked into more carefully.

Reply:
"Given event files are consumed not just by spark, but external applications and analysis tools" sounds reasonable, i'll remove the changes, follow this comment, how about current

Reply:
Thanks @summaryzb !
Note - if there are good reasons to change, I am all for it !